.github/workflows/codeql-analysis.yml

```diff
@@ -19,6 +19,10 @@ jobs:
     name: Analyze
     runs-on: ubuntu-latest
 
+    permissions:
+      # required for all workflows
+      security-events: write
```

Comment:
I don't think that will work without a `permissions: read-all` at top level, will it ?
(if any specific permission is set, anything that is unset defaults to `none`).

I guess this doesn't come up in the PR's checks because CodeQL is only activated on main branch 🤔

Reply:
I added a top level permissions